tools/upgrade/config.py

```diff
@@ -0,0 +1,349 @@
+"""Config upgrade tool.
```

Comment:
How long do we plan to support the upgrade? Maybe we may have to add tests to ensure that this works down the line. Can can consider adding two configs as reference and target, and create tests similar to https://github.com/openvinotoolkit/anomalib/blob/main/tests/integration/tools/test_lightning_entrypoint.py

Reply:
I agree that a unit test would be useful. In addition, I think it would be good to add a tutorial how to use this tool in the documentation.

Reply:
> I agree that a unit test would be useful. In addition, I think it would be good to add a tutorial how to use this tool in the documentation.

Added in e60f971e692ac86c1c2cf1b9abc79195761e8c00

Reply:
> How long do we plan to support the upgrade? Maybe we may have to add tests to ensure that this works down the line. Can can consider adding two configs as reference and target, and create tests similar to https://github.com/openvinotoolkit/anomalib/blob/main/tests/integration/tools/test_lightning_entrypoint.py

Added in e60f971e692ac86c1c2cf1b9abc79195761e8c00